code/game/objects/items/flamethrower.dm

```diff
@@ -230,28 +238,94 @@
 		igniter.secured = FALSE
 		status = TRUE
 		if(create_with_tank)
-			ptank = new /obj/item/tank/internals/plasma/full(src)
+			beaker = new /obj/item/reagent_containers/glass/beaker/large(src)
+			beaker.reagents.add_reagent(/datum/reagent/fuel, beaker.reagents.maximum_volume)
 		update_icon()
 
 /obj/item/flamethrower/full
+	icon = 'icons/obj/guns/48x32guns.dmi'
+	item_state = "prebuilt_flamethrower_0"
 	create_full = TRUE
 
+/obj/item/flamethrower/full/update_icon_state()
+	item_state = "prebuilt_flamethrower_[lit]"
+
 /obj/item/flamethrower/full/tank
 	create_with_tank = TRUE
 
 /obj/item/flamethrower/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
 	var/obj/projectile/P = hitby
-	if(damage && attack_type == PROJECTILE_ATTACK && P.damage_type != STAMINA && prob(15))
+	if(beaker && damage && attack_type == PROJECTILE_ATTACK && P.damage_type != STAMINA && prob(15))
 		owner.visible_message("<span class='danger'>\The [attack_text] hits the fueltank on [owner]'s [name], rupturing it! What a shot!</span>")
 		var/turf/target_turf = get_turf(owner)
 		log_game("A projectile ([hitby]) detonated a flamethrower tank held by [key_name(owner)] at [COORD(target_turf)]")
-		igniter.ignite_turf(src,target_turf, release_amount = 100)
-		qdel(ptank)
+		var/turf/flamer_turf = get_turf(owner)
+		flamer_turf.IgniteTurf(30)
+		QDEL_NULL(beaker)
 		return 1 //It hit the flamethrower, not them
 
+///FLAMETHROWER PROJECTILE
+/obj/projectile/flamethrower
+	name = "\the flames"
+	damage = 0
+	speed = 1
+	hitsound = ""
+	icon_state = null
+	icon = null
+
+	///the turf the target was on when we clicked the flamethrower
+	var/turf_target
+	///the color the turf fire will be when igniting a turf
+	var/flame_color = "red"
+	///the flamethrower this was shot from
+	var/obj/item/flamethrower/linked_flamethrower
+	///how much power do we give the turf fire once it ignites?
+	var/power = 4
+	/// how much do we subtract from the beaker?
+	var/release_amount = 8
 
-/obj/item/assembly/igniter/proc/flamethrower_process(turf/open/location)
-	location.hotspot_expose(700,2)
 
-/obj/item/assembly/igniter/proc/ignite_turf(obj/item/flamethrower/F,turf/open/location,release_amount = 0.05)
-	F.default_ignite(location,release_amount)
+/obj/projectile/flamethrower/Move()
+	. = ..()
+	if(!linked_flamethrower)
+		return
+	if(!linked_flamethrower.beaker)
+		return
+	var/turf/location = get_turf(src)
+
+	var/datum/reagents/beaker_reagents = linked_flamethrower.beaker.reagents
+	var/datum/reagents/my_fraction = new()
+
+	beaker_reagents.trans_to(my_fraction, release_amount, no_react = TRUE)
```

Comment:
don't link the flamethrower to it, just transfer the reagents when the projectile is first created by the flamethrower